spotlight-client/src/contentModels/RecidivismRateMetric.test.ts

```diff
@@ -73,3 +73,93 @@ describe("single followup period", () => {
     expect.hasAssertions();
   });
 });
+
+describe("cohorts data series", () => {
+  let metric: RecidivismRateMetric;
+
+  const testMetricId = "PrisonRecidivismRateHistorical";
+
+  afterEach(() => {
+    metric.setSelectedCohorts(undefined);
+  });
+
+  test("all cohorts", async () => {
+    metric = await getPopulatedMetric(testMetricId);
+
+    reactImmediately(() => {
+      expect(metric.cohortDataSeries).toMatchSnapshot();
+    });
+
+    expect.hasAssertions();
+  });
+
+  test("selected cohorts", async () => {
+    metric = await getPopulatedMetric(testMetricId);
+
+    metric.setSelectedCohorts([2017, 2018]);
+
+    reactImmediately(() => {
+      // this should be a subset of the all cohorts snapshot;
+      // not only the records but also the series colors should be identical
+      expect(metric.cohortDataSeries).toMatchSnapshot();
+    });
+
+    expect.hasAssertions();
+  });
+
+  test.each(["raceOrEthnicity", "gender", "ageBucket"] as const)(
+    "single cohort with %s views",
+    async (demographicView) => {
+      metric = await getPopulatedMetric(testMetricId);
+
+      metric.setSelectedCohorts([2017]);
+
+      runInAction(() => {
+        metric.demographicView = demographicView;
+      });
+
+      reactImmediately(() => {
+        expect(metric.cohortDataSeries).toMatchSnapshot();
+      });
+
+      // reset demographics
+      runInAction(() => {
+        metric.demographicView = "total";
+      });
+
+      expect.hasAssertions();
+    }
+  );
+
+  test("demographic view resets to total when multiple cohorts are selected", async () => {
```

Comment:
Just a note about this. I found it a bit confusing about when the demographic dropdown was enable vs disabled - and what to do to enable it again. Maybe a tooltip explaining that the demographic view is enabled when a single cohort is selected would be helpful?

Reply:
good catch! In v1 there was actually a note about this in the section body, which I somehow managed to drop in the v2 transition -- I'll restore that before merging